math/src/main/java/com/powsybl/math/timeseries/TimeSeriesTable.java

```diff
@@ -141,49 +164,59 @@ private void checkIndex(TimeSeries timeSeries) {
-            int i = stringTimeSeriesNames.add(timeSeries.getMetadata().getName());
-            timeSeriesIndexDoubleOrString.add(i);
-        }
+            for (DoubleTimeSeries timeSeries : doubleTimeSeries) {
+                checkIndex(timeSeries);
+                timeSeriesMetadata.add(timeSeries.getMetadata());
+                int i = doubleTimeSeriesNames.add(timeSeries.getMetadata().getName());
+                timeSeriesIndexDoubleOrString.add(i);
+            }
+            for (StringTimeSeries timeSeries : stringTimeSeries) {
+                checkIndex(timeSeries);
+                timeSeriesMetadata.add(timeSeries.getMetadata());
+                int i = stringTimeSeriesNames.add(timeSeries.getMetadata().getName());
+                timeSeriesIndexDoubleOrString.add(i);
+            }
 
-        if (tableIndex == null) {
-            throw new TimeSeriesException("None of the time series have a finite index");
-        }
+            if (tableIndex == null) {
+                throw new TimeSeriesException("None of the time series have a finite index");
+            }
+
+            int versionCount = toVersion - fromVersion + 1;
 
-        int versionCount = toVersion - fromVersion + 1;
+            // allocate double buffer
+            int doubleBufferSize = versionCount * doubleTimeSeriesNames.size() * tableIndex.getPointCount();
+            doubleBuffer = createDoubleBuffer(doubleBufferSize, Double.NaN);
 
-        // allocate double buffer
-        int doubleBufferSize = versionCount * doubleTimeSeriesNames.size() * tableIndex.getPointCount();
-        doubleBuffer = createDoubleBuffer(doubleBufferSize, Double.NaN);
+            // allocate string buffer
+            int stringBufferSize = versionCount * stringTimeSeriesNames.size() * tableIndex.getPointCount();
+            stringBuffer = new CompactStringBuffer(stringBufferSize);
+
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info("Allocation of {} for time series table",
+                        FileUtils.byteCountToDisplaySize((long) doubleBuffer.capacity() * Double.BYTES + stringBuffer.capacity() * Integer.BYTES));
+            }
 
-        // allocate string buffer
-        int stringBufferSize = versionCount * stringTimeSeriesNames.size() * tableIndex.getPointCount();
-        stringBuffer = new CompactStringBuffer(stringBufferSize);
+            // allocate statistics buffer
+            means = new double[doubleTimeSeriesNames.size() * versionCount];
+            Arrays.fill(means, Double.NaN);
 
-        if (LOGGER.isInfoEnabled()) {
-            LOGGER.info("Allocation of {} for time series table",
-                    FileUtils.byteCountToDisplaySize((long) doubleBuffer.capacity() * Double.BYTES + stringBuffer.capacity() * Integer.BYTES));
+            stdDevs = new double[doubleTimeSeriesNames.size() * versionCount];
+            Arrays.fill(stdDevs, Double.NaN);
+        } finally {
```

Comment:
Add a catch statement to reset timeSeriesMetadata, doubleBuffer and stringBuffer to null?